lib/backend/client.go

```diff
@@ -62,5 +62,5 @@ type Client interface {
 	Download(namespace, name string, dst io.Writer) error
 
 	// List lists entries whose names start with prefix.
-	List(prefix string) ([]string, error)
+	List(prefix string, options *ListOptions) ([]string, string, error)
```

Comment:
would defining ListResult better? That way you can include the nextToken. Also, why pointer?

Reply:
The pointer was so callers can pass `nil` or `&ListOptions{}` if they don't want to set any options and the backends will handle both the same way.

I agree defining `ListResult` would probably be better if there are other things we may want to include in the response in the future. But I'm not sure how much better it is if there's only two fields we want to return.

Reply:
it can also be `ListOptions{}` and the backend would interpret the default. Another way to do this is to use `List(prefix string, opts...ListOptions)`.

I think I prefer `ListResult` more. It is similar to `listOutput, err := c.s3.ListObjectsV2(listInput)` and you dont need to add `return x,"",err` to other backend implementations.